src/Cardano/Transaction.cpp

```diff
@@ -196,14 +216,21 @@ Cbor::Encode cborizeOutputAmounts(const Amount& amount, const TokenBundle& token
         return Cbor::Encode::uint(amount);
     }
     // native and token amounts
-    std::vector<pair<Cbor::Encode, Cbor::Encode>> tokensMap;
-    for (auto iter = tokenBundle.bundle.begin(); iter != tokenBundle.bundle.end(); ++iter) {
-        tokensMap.push_back(make_pair(
-            Cbor::Encode::bytes(parse_hex(iter->second.policyId)),
-            Cbor::Encode::map({make_pair(
-                Cbor::Encode::bytes(data(iter->second.assetName)),
-                Cbor::Encode::uint(uint64_t(iter->second.amount)) // 64 bits
-            )})
+    // tokens: organized in two levels: by policyId and by assetName
+    const auto policyIds = tokenBundle.getPolicyIds();
+    map<Cbor::Encode, Cbor::Encode> tokensMap;
+    for (const auto& policy: policyIds) {
+        const auto& subTokens = tokenBundle.getByPolicyId(policy);
+        map<Cbor::Encode, Cbor::Encode> subTokensMap;
+        for (const auto& token: subTokens) {
+            subTokensMap.emplace(make_pair(
+                Cbor::Encode::bytes(data(token.assetName)),
+                Cbor::Encode::uint(uint64_t(token.amount)) // 64 bits
+            ));
+        }
+        tokensMap.emplace(make_pair(
+            Cbor::Encode::bytes(parse_hex(policy)),
+            Cbor::Encode::map(subTokensMap)
         ));
```

Comment:
Same question regarding if we need the map to be ordered here.

since we use [emplace](https://en.cppreference.com/w/cpp/container/map/emplace) `make_pair` is not required, it will just use the template argument constructor